axolotl-web/src/router/router.js

```diff
@@ -96,42 +102,47 @@ export const router = new Router({
 });
 
 router.beforeEach((to, from, next) => {
+
   if (to.path === "/debug") {
     return next();
   }
 
-  const registrationStatus = localStorage.getItem('registrationStatus');
-
-  // If we're going to the waiting status page
-  if (to.path === "/waitForRegistrationStatus") {
-    // And we still don't have the registration status of the user
-    if (registrationStatus == null) {
-      // Then we go to the registration page
-      return next();
-    } else {
-      // Else we go to the home
-      return next("/");
-    }
-  }
-
-  // We're not requesting the registration page.
-  // But we should do so if we don't have registration status yet, let's redirect the user
-  if (registrationStatus == null) {
-    // The information about logging isn't there yet, go to the waiting page
-    return next('/waitForRegistrationStatus');
+  console.log("1 " + store.state.isRegistrationStatusUpToDate + " / " + store.state.registrationStatus);
+  if (!store.state.isRegistrationStatusUpToDate) {
+    store.dispatch("getRegistrationStatus");
+    store.watch((state) => state.isRegistrationStatusUpToDate, function() {
+      console.log("2 " + store.state.isRegistrationStatusUpToDate + " / " + store.state.registrationStatus);
+      proceed(to, next);
+    });
+  } else {
+    console.log("3 " + store.state.isRegistrationStatusUpToDate + " / " + store.state.registrationStatus);
+    proceed(to, next);
   }
+});
 
-  // We have a status (regisered or not). Let's check access right
-  const publicPages = ['/register', '/verify', '/password'];
-  const authRequired = !publicPages.includes(to.path);
+function proceed(to, next) {
+  const registrationPages = ['/register', '/verify', '/password', '/pin'];
+  const registrationStatus = store.state.registrationStatus;
+  store.isRegistrationStatusUpToDate = false;
```

Comment:
I don't know where to put this. Leaving it there means a call to the server will be done each time we request a route.

Reply:
Watching the `SET_REGISTRATION_STATUS` in the vuex store, it's only sent twice from the backend. No matter how many chats or settings pages I open.

Reply:
Yes there is something not working as I expect there, that's also probably the reason why I got stuck once. Do I need to "commit" or do something like this to actually change the value in the store?

Reply:
yup, vuex uses the flux principle and therefore the database/vuex-storage works like git. Every change has to be committed.

Reply:
You need a mutation in store.js like:
```
    SET_REGESTRATION_STATUS_UP_TO_DATE(state, status) {
      store.isRegistrationStatusUpToDate = status;
    },
```
and then you can do something like:
```
store.commit("SET_REGESTRATION_STATUS_UP_TO_DATE", false)
```